python/lsst/ip/isr/photodiode.py

```diff
@@ -0,0 +1,260 @@
+        Log to write messages to.
+    **kwargs :
+        Paramters to pass to parent constructor.
+    """
+
+    _OBSTYPE = 'PHOTODIODE'
+    _SCHEMA = 'Gen3 Photodiode'
+    _VERSION = 1.0
+
+    def __init__(self, timeSamples=None, currentSamples=None, **kwargs):
+        if timeSamples is not None and currentSamples is not None:
+            if len(timeSamples) != len(currentSamples):
+                raise RuntimeError(f"Inconsitent vector lengths: {len(timeSamples)} {len(currentSamples)}")
+            else:
+                self.timeSamples = np.array(timeSamples)
+                self.currentSamples = np.array(currentSamples)
+        else:
+            self.timeSamples = np.array([])
+            self.currentSamples = np.array([])
+
+        super().__init__(**kwargs)
+
+        if 'integrationMethod' in kwargs:
+            self.integrationMethod = kwargs.pop('integrationMethod')
+        else:
+            self.integrationMethod = 'DIRECT_SUM'
+
+        if 'day_obs' in kwargs:
+            self.updateMetadata(day_obs=kwargs['day_obs'])
+        if 'seq_num' in kwargs:
+            self.updateMetadata(seq_num=kwargs['seq_num'])
+
+        self.requiredAttributes.update(['timeSamples', 'currentSamples', 'integrationMethod'])
+
+    @classmethod
+    def fromDict(cls, dictionary):
+        """Construct a PhotodiodeCalib from a dictionary of properties.
+
+        Parameters
+        ----------
+        dictionary : `dict`
+            Dictionary of properties.
+
+        Returns
+        -------
+        calib : `lsst.ip.isr.PhotodiodeCalib`
+            Constructed photodiode data.
+
+        Raises
+        ------
+            Raised if the supplied dictionary is for a different
+            calibration type.
+        """
+        calib = cls()
+
+        if calib._OBSTYPE != dictionary['metadata']['OBSTYPE']:
+            raise RuntimeError(f"Incorrect photodiode supplied.  Expected {calib._OBSTYPE}, "
+                               f"found {dictionary['metadata']['OBSTYPE']}")
+
+        calib.setMetadata(dictionary['metadata'])
+
+        calib.timeSamples = np.array(dictionary.get('timeSamples'))
+        calib.currentSamples = np.array(dictionary.get('currentSamples'))
+        calib.integrationMethod = dictionary.get('integrationMethod')
+
+        calib.updateMetadata()
+        return calib
+
+    def toDict(self):
+        """Return a dictionary containing the photodiode properties.
+
+        The dictionary should be able to be round-tripped through.
+        `fromDict`.
+
+        Returns
+        -------
+        dictionary : `dict`
+            Dictionary of properties.
+        """
+        self.updateMetadata()
+
+        outDict = {}
+        metadata = self.getMetadata()
+        outDict['metadata'] = metadata
+
+        outDict['timeSamples'] = self.timeSamples.tolist()
+        outDict['currentSamples'] = self.currentSamples.tolist()
+
+        outDict['integrationMethod'] = self.integrationMethod
```

Comment:
Now this seems like an instance for a `.get()`, especially as, if I'm reading it right, the `fromDict()` method isn't dealing with the `integrationMethod`, or is it?